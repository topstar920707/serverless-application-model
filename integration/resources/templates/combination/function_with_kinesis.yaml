Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        KinesisStream:
          Type: Kinesis
          Properties:
            Stream:
              # Connect with the stream we have created in this template
              Fn::GetAtt: [MyStream, Arn]

            BatchSize: 100
            StartingPosition: LATEST
            TumblingWindowInSeconds: 120
            FunctionResponseTypes:
              - ReportBatchItemFailures

  MyStream:
    Type: AWS::Kinesis::Stream
    Properties:
      ShardCount: 1
