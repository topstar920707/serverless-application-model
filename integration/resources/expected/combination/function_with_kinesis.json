[
  {
    "LogicalResourceId": "MyLambdaFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyLambdaFunction2",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyLambdaFunction2Role",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyStream",
    "ResourceType": "AWS::Kinesis::Stream"
  },
  {
    "LogicalResourceId": "MyLambdaFunction2KinesisStream",
    "ResourceType": "AWS::Lambda::EventSourceMapping"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionKinesisStream",
    "ResourceType": "AWS::Lambda::EventSourceMapping"
  }
]
